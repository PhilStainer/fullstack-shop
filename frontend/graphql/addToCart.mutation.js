import gql from 'graphql-tag'

const ADD_TO_CART_MUTATION = gql`
  mutation($id: ID!) {
    addToCart(id: $id) {
      _id
      quantity
      item {
        _id
        title
        description
        imageUrl
        price
      }
    }
  }
`

export default ADD_TO_CART_MUTATION
