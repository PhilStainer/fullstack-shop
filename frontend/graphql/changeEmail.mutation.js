import gql from 'graphql-tag'

const CHANGE_PASSWORD_MUTATION = gql`
  mutation($password: String!, $email: String!, $confirmEmail: String!) {
    changeEmail(
      password: $password
      email: $email
      confirmEmail: $confirmEmail
    ) {
      status
      message
    }
  }
`

export default CHANGE_PASSWORD_MUTATION
