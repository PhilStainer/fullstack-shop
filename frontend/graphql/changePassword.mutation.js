import gql from 'graphql-tag'

const CHANGE_PASSWORD_MUTATION = gql`
  mutation(
    $currentPassword: String!
    $password: String!
    $confirmPassword: String!
  ) {
    changePassword(
      currentPassword: $currentPassword
      password: $password
      confirmPassword: $confirmPassword
    ) {
      status
      message
    }
  }
`

export default CHANGE_PASSWORD_MUTATION
