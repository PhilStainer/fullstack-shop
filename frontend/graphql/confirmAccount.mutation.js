import gql from 'graphql-tag'

const CONFIRM_ACCOUNT_MUTATION = gql`
  mutation($confirmToken: String!) {
    confirmAccount(confirmToken: $confirmToken) {
      status
      message
    }
  }
`

export default CONFIRM_ACCOUNT_MUTATION
