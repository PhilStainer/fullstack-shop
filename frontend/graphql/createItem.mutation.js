import gql from 'graphql-tag'

const CREATE_ITEM_MUTATION = gql`
  mutation(
    $title: String!
    $description: String!
    $imageUrl: String!
    $price: Int!
  ) {
    createItem(
      title: $title
      description: $description
      imageUrl: $imageUrl
      price: $price
    ) {
      _id
    }
  }
`

export default CREATE_ITEM_MUTATION
