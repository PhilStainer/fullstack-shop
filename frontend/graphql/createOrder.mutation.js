import gql from 'graphql-tag'

const CREATE_ORDER_MUTATION = gql`
  mutation($token: String!) {
    createOrder(token: $token) {
      _id
    }
  }
`

export default CREATE_ORDER_MUTATION
