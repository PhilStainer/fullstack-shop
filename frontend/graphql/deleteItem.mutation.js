import gql from 'graphql-tag'

const DELETE_ITEM_MUTATION = gql`
  mutation($id: ID!) {
    deleteItem(id: $id) {
      _id
    }
  }
`

export default DELETE_ITEM_MUTATION
