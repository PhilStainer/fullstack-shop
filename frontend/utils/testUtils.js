export const fakeUser = () => ({
  _id: '12345',
  name: 'Test Name',
  email: 'test@email.com',
})
