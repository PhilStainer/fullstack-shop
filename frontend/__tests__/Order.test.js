import {render, waitFor} from '@testing-library/react'
import {MockedProvider} from '@apollo/react-testing'
import {GraphQLError} from 'graphql'

import ORDER_QUERY from '#root/graphql/order.query'

import Order from '#root/pages/Order'
import {fakeOrder} from '#root/utils/testUtils'

const query = {id: 'order id'}

test('fetch items on mount', async () => {
  const {getByText} = render(
    <MockedProvider>
      <Order query={query} />
    </MockedProvider>,
  )

  await waitFor(() => {
    expect(getByText(/loading/i)).toBeInTheDocument()
  })
})

test('renders error on graphql error', async () => {
  const mock = {
    request: {
      query: ORDER_QUERY,
      variables: {id: query.id},
    },
    result: jest.fn(() => ({
      errors: [new GraphQLError('Error finding order...')],
    })),
  }

  const {getByText} = render(
    <MockedProvider mocks={[mock]} addTypename={false}>
      <Order query={query} />
    </MockedProvider>,
  )

  await waitFor(() => {
    expect(getByText(/error finding order/i)).toBeInTheDocument()
  })
})

test('renders order', async () => {
  const mock = {
    request: {
      query: ORDER_QUERY,
      variables: {id: query.id},
    },
    result: jest.fn(() => ({
      data: {order: fakeOrder()},
    })),
  }

  const {getByTestId} = render(
    <MockedProvider mocks={[mock]} addTypename={false}>
      <Order query={query} />
    </MockedProvider>,
  )

  await waitFor(() => {
    expect(getByTestId('order')).toMatchSnapshot()
  })
})
