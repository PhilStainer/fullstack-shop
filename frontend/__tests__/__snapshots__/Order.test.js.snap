// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders order 1`] = `
<div
  class="StyledOrder-dj9zr0-0 kgBjuJ"
  data-testid="order"
>
  <p>
    <span>
      Order ID:
    </span>
    <span>
      12345
    </span>
  </p>
  <p>
    <span>
      Charge
    </span>
    <span>
      ch_12345
    </span>
  </p>
  <p>
    <span>
      Date
    </span>
    <span>
      April 5, 2018 8:24 pm
    </span>
  </p>
  <p>
    <span>
      Order Total
    </span>
    <span>
      £50
    </span>
  </p>
  <p>
    <span>
      Item Count
    </span>
    <span>
      1
    </span>
  </p>
  <div
    class="items"
  >
    <div
      class="order-item"
    >
      <img
        alt="Title"
        src="https://localhost:3000/"
      />
      <div
        class="item-details"
      >
        <h2>
          Title
        </h2>
        <p>
          1
           x 
          £99.99
           =
           
          £99.99
        </p>
        <p>
          Description
        </p>
      </div>
    </div>
  </div>
</div>
`;
